### PR TITLE
add testing web clients from code

### PR DESCRIPTION
**Pull request by felix920506**

### CODE REVIEW

File: docs/general/testing/web/index.md
Comment (Gauvino) [inline docs/general/testing/web/index.md:32]:
```suggestion
Jellyfin Web can be hosted as a standalone application without being associated to a Jellyfin server. In that case, a separate Jellyfin server will need to be prepared.
```

**Review by Gauvino — CHANGES_REQUESTED** [review]

**Comment by felix920506** [comment]
I don't know if the added instructions should be on their own page or in the same page as testing from the cloudflare pages deployments

**Comment by ferferga** [comment]
Vue has also their own docs for development, maybe a good idea to link them in this section: https://github.com/jellyfin/jellyfin-vue/wiki/Contributing#development-setup

File: docs/general/testing/web/index.md
Comment (ferferga) [inline docs/general/testing/web/index.md:82]:
```suggestion
More info on how to setup Jellyfin Vue in a development environemt can be found [here](https://github.com/jellyfin/jellyfin-vue/wiki/Contributing#development-setup)
```
```suggestion
More info on how to setup Jellyfin Vue in a development environmet can be found [here](https://github.com/jellyfin/jellyfin-vue/wiki/Contributing#development-setup)
```

File: docs/general/testing/web/index.md
Comment (Gauvino) [inline docs/general/testing/web/index.md:32]:
```suggestion
Jellyfin web clients can be hosted as a standalone application without being associated with a Jellyfin server. In that case, a separate Jellyfin server will need to be prepared.
```

File: docs/general/testing/web/index.md
Comment (Gauvino) [inline docs/general/testing/web/index.md:38]:
```suggestion
Below is a list of things to prepare before testing Jellyfin web clients.
```

File: docs/general/testing/web/index.md
Comment (Gauvino) [inline docs/general/testing/web/index.md:57]:
```suggestion
The `master` branch is where the unstable build is built from. If you wish to test over there, please checkout that branch. To checkout the `master` branch, simply do the following.
```

File: docs/general/testing/web/index.md
Comment (Gauvino) [inline docs/general/testing/web/index.md:60]:
```suggestion
- GitHub Desktop: Select the `master` branch in the branch dropdown menu.
```

File: docs/general/testing/web/index.md
Comment (Gauvino) [inline docs/general/testing/web/index.md:73]:
```suggestion
3. Run `npm install` to install the prerequisites. This will need to be done every time you switch to a new branch.
```

File: docs/general/testing/web/index.md
Comment (Gauvino) [inline docs/general/testing/web/index.md:78]:
```suggestion
2. Run `npm start` to start a development server. Allow firewall access when prompted if you are using Windows. The server will start on port 8080 for Jellyfin Web and Port 3000 for Jellyfin Vue. The server will auto update if the contents of the folder are changed.
```

**Review by Gauvino — CHANGES_REQUESTED** [review]
Typo fix

File: docs/general/testing/web/index.md
Comment (ferferga) [inline docs/general/testing/web/index.md:32]:
```suggestion
Jellyfin web clients can be hosted as a standalone application without being associated with a Jellyfin server. In that case, a separate Jellyfin server will be needed.
```
I'm not sure that second sentence it's necessary as well. The first one is clear enough and I feel like the second one could be misinterpreted, leading people to think that at least 2 servers are needed somehow?

File: docs/general/testing/web/index.md
Comment (ferferga) [inline docs/general/testing/web/index.md:44]:
```suggestion
- (Optional) A text editor or web IDE to make changes during testing, eg. [VSCode](https://code.visualstudio.com/), [Notepad++](https://notepad-plus-plus.org/), [Jetbrains Webstorm](https://www.jetbrains.com/webstorm/).
```

File: docs/general/testing/web/index.md
Comment (ferferga) [inline docs/general/testing/web/index.md:62]: Let's use Pull Request from the beginning here

File: docs/general/testing/web/index.md
Comment (ferferga) [inline docs/general/testing/web/index.md:73]: I'd remove this section and link to the documentation of Web and Vue, since this is perhaps the section more likely to be changed in the future (while the others are less likely to). Things like setting up a monorepo, changing package manager, etc... Will outdate this easily. This, in fact, is not applicable to Vue (you must `cd frontend` first).

I'm also not sure the next one is needed as well for the same reasons. Only a small port change (e.g: Web switches to Vite and the new port is used instead of the old one) and this is out of date.

All this maybe could be solved by using a sentence similar to the last one applicable for Vue, but for web, linking to the repo where info is probably going to be more up to date.

I leave this up to you/other reviewers though. I also understand my concerns are more applicable to Vue in this case (since it usually live sin the bleeding edge) than Web.

**Review by ferferga — CHANGES_REQUESTED** [review]

**Review by ferferga — APPROVED** [review]
LGTM now!

**Comment by felix920506** [comment]
dealing with merge conflicts

File: docs/general/testing/web/index.md
Comment (thornbill) [inline docs/general/testing/web/index.md:77]: The capitalization of the headings is inconsistent. Are we only capitalizing the first letter of the first word or every word?

File: docs/general/testing/web/index.md
Comment (thornbill) [inline docs/general/testing/web/index.md:75]:
```suggestion
2. Checkout the pull request in Git: [Git CLI and GitHub CLI](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/checking-out-pull-requests-locally), [GitHub Desktop](https://docs.github.com/en/desktop/working-with-your-remote-repository-on-github-or-github-enterprise/viewing-a-pull-request-in-github-desktop).
```

**Review by thornbill — CHANGES_REQUESTED** [review]

File: docs/general/testing/web/index.md
Comment (ferferga) [reply docs/general/testing/web/index.md]: Vue is strictly current LTS as well, yes.

File: docs/general/testing/web/index.md
Comment (thornbill) [inline docs/general/testing/web/index.md:51]: lgtm after this :rocket: 

```suggestion
- (Optional) Install nvm: [macOS, Linux](https://github.com/nvm-sh/nvm?tab=readme-ov-file#installing-and-updating), [Windows](https://github.com/coreybutler/nvm-windows).
```

**Review by thornbill — APPROVED** [review]

**Pull request merged**
